sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -488,6 +488,8 @@ object DataType {
 
 /**
  * Jackson serializer for [[DataType]]. Internally this delegates to json4s based serialization.
+ *
+ * @since 3.0.0
  */
 class DataTypeJsonSerializer extends JsonSerializer[DataType] {
```

Comment:
these 2 were added in https://github.com/apache/spark/pull/26127/files#diff-df78a74ef92d9b8fb4ac142ff9a62464R490

I don't think we want to publish them. Can we move them to a private package? like `xxx.catalyst.utils`
also cc @hvanhovell

Reply:
Good catch. We should definitely move it into catalyst. I am not sure if we can change the visibility (jackson might not like that), but we should just try.

Reply:
classes in the catalyst package are treated as private classes and we don't really need to add the `private[sql]`

Reply:
Thanks, move them to `org.apache.spark.sql.catalyst.util.DataTypeJsonUtils`.